### PR TITLE
Fix docs (README.rst and setup.rst)

### LINKED ISSUE #584: make releases clear
At the moment the "latest" release on GH is quite old, older than 1.2.0 (advertised on mpmath.org as "latest")

And what is the status of 1.2.1? Is it the real latest? A pre-release?

### PR DESCRIPTION
**Pull request by skirpichev**
I hope, this finally fixes #584.  (Another part goes to  #589)

@rocky, I'll appreciate your review.

### CODE REVIEW

**Comment by rocky** [comment]
@skirpichev On performing a summary lookover,  these changes look very useful and  helpful. 

However I would like to go over this in detail when I get a chance. 

Thanks for doing.

**Comment by skirpichev** [comment]
On Fri, Apr 23, 2021 at 11:47:30PM -0700, R. Bernstein wrote:
>    However I would like to go over this in detail when I get a chance.

Thank you.  I'll wait.

File: docs/setup.rst [hunk 1]
```diff
@@ -73,55 +78,47 @@ After the setup has completed, you should be able to fire up the interactive Pyt
...
+
+    If you have are upgrading mpmath from an earlier version, you may have to manually uninstall the old version or remove the old files.
 
 Using gmpy (optional)
 ---------------------
 
-By default, mpmath uses Python integers internally. If `gmpy <http://code.google.com/p/gmpy/>`_ version 1.03 or later is installed on your system, mpmath will automatically detect it and transparently use gmpy integers intead. This makes mpmath much faster, especially at high precision (approximately above 100 digits).
-
-To verify that mpmath uses gmpy, check the internal variable ``BACKEND`` is not equal to 'python':
+By default, mpmath uses Python integers internally. If `gmpy <https://github.com/aleaxit/gmpy>`_ version 2.1.0a4 or later is installed on your system, mpmath will automatically detect it and transparently use gmpy integers intead. This makes mpmath much faster, especially at high precision (approximately above 100 digits).
```
Comment (rocky) [inline docs/setup.rst:88]: intead -> instead

**Comment by rocky** [comment]
Some observations. 

It took me a while to figure out how to build the docs (even though the instructions _are_ in diff of of this PR!) .  Personally, I think it would be helpful to add somewhere outside of the sel-referential sphinx doc itself how to build the docs. 

Some possibilities include  a `README` in the doc directory with a non`.rst` extension (e.g. no extention or a `.txt` or `.md` extension) or at the toplevel `README.rst`.  Another thing that I typically do is create a Makefile and add the targets there. https://github.com/Mathics3/mathics-development-guide/blob/master/docs/Makefile is an example of this. 

When finally realized what I needed to do to build the sphinx doc, I realized I didn't have `sphinx` installed (for that particular version of Python).  So it would be good to have a `requirements-dev.txt` at the top-level or mention `pip install sphinx` somewhere. 

I found the instruction

```
pip install -e [develop,docs]
```

confusing.  From the clone directory. `pip install -e develop` works, but `pip install -e docs` didn't for me. And of course `pip install -e [develop,docs]` is erroneous. 

Running `setup.rst` over a spelling checker, "intead" -> "instead".  (noted above)

The good news though is that when I finally was able to install build the docs sphinx built the docs without any warnings or errors.  Similarly, running `pytest --pyargs mpmath` (or `py.test ...`) worked without any errors.

In the mpmath.org page which doesn't appear in the PR or repository, feel free to add `https://mathics.org` under software that uses mpmath.

**Comment by rocky** [comment]
Technically the "=" line in line two of https://raw.githubusercontent.com/fredrik-johansson/mpmath/master/docs/references.rst should be shorter. I am not sure why sphinx didn't mention it. It usually does for me.

**Comment by skirpichev** [comment]
> Personally, I think it would be helpful to add somewhere outside of the self-referential sphinx doc itself how to build the docs.

Something "outside of the docs" will require some stuff in the docs and efforts to keep this in sync.  I wouldn't buy that.

> I realized I didn't have sphinx installed

That should be handled by the "current development version" section in the file.  Probably, this should be in the "Compiling the documentation".  Thank you, I'll think about.

> I found the instruction pip install -e [develop,docs] confusing.

It seems to be your fault).  I thought it was a typo, but it wasn't.  Use `pip install -e .[develop,docs] ` instead.  The dot is important, see the pip help.

> Running setup.rst over a spelling checker, "intead" -> "instead". (noted above)

Thanks, fixed.  BTW, there are sphinxcontrib-spelling extension.  Maybe it can be used for mpmath.

> In the mpmath.org page which doesn't appear in the PR or repository

Do you mean that the repo references http (not https) page?  I can't change this.

> feel free to add https://mathics.org under software that uses mpmath.

Again, this part can only be fixed by Fredrik.

> Technically the "=" line in line two of https://raw.githubusercontent.com/fredrik-johansson/mpmath/master/docs/references.rst should be shorter.

Oh, yes.  The important part: it's long enough to keep formatting right.

> It usually does for me.

How it can be?

Thank you for more review.

**Pull request closed without merging**